src/Storages/System/StorageSystemNumbers.cpp

```diff
@@ -46,77 +55,125 @@ class NumbersSource : public ISource
     UInt64 block_size;
     UInt64 next;
     UInt64 step;
-
-    static Block createHeader()
-    {
-        return { ColumnWithTypeAndName(ColumnUInt64::create(), std::make_shared<DataTypeUInt64>(), "number") };
-    }
 };
 
-
-struct NumbersMultiThreadedState
-{
-    std::atomic<UInt64> counter;
-    explicit NumbersMultiThreadedState(UInt64 offset) : counter(offset) {}
-};
-
-using NumbersMultiThreadedStatePtr = std::shared_ptr<NumbersMultiThreadedState>;
-
-class NumbersMultiThreadedSource : public ISource
+/// Generate numbers according to ranges.
+/// Numbers generated is ordered in one stream.
+/// And we will not generate additional numbers out of ranges.
+class NumbersRangedSource : public ISource
 {
 public:
-    NumbersMultiThreadedSource(NumbersMultiThreadedStatePtr state_, UInt64 block_size_, UInt64 max_counter_)
-        : ISource(createHeader())
-        , state(std::move(state_))
-        , block_size(block_size_)
-        , max_counter(max_counter_) {}
-
-    String getName() const override { return "NumbersMt"; }
```

Comment:
Please search for `NumbersMt` and update the necessary docs. I found two mentions of this:
1. `docs/en/sql-reference/statements/explain.md`
2. `docs/ru/sql-reference/statements/explain.md`

Reply:
I really ignored docs, thanks.  Do I need update docs other than `en` directory?

Reply:
I would say only if the change are renaming trivial stuff (like `NumbersMt` to `NumbersRange`).